pkg/ipam/node_manager.go

```diff
@@ -288,6 +294,16 @@ func (n *NodeManager) Update(resource *v2.CiliumNode) (nodeSynced bool) {
 			logLimiter:          logging.NewLimiter(10*time.Second, 3), // 1 log / 10 secs, burst of 3
 		}
 
+		// InstanceAPI is stale and the instances API is stable then do resync instancesAPI to sync instances
+		if !n.instancesAPI.HasInstance(resource.InstanceID()) && n.stableInstancesAPI {
+			if syncTime := n.instancesAPI.Resync(context.TODO()); syncTime.IsZero() {
+				node.logger().Warning("Failed to resync the instances from the API after new node was found")
+				n.stableInstancesAPI = false
+			} else {
+				n.stableInstancesAPI = true
+			}
+		}
+
 		node.ops = n.instancesAPI.CreateNode(resource, node)
 
 		ctx, cancel := context.WithCancel(context.Background())
```

Comment:
One last request: Do you mind moving this up and pass `ctx` it into `n.instancesAPI.Resync`? It doesn't affect correctness (since the context won't be cancelled till later), but if we ever change the behavior of `Resync` to spawn any go routines, it will ensure that those are cancelled when the node is removed.

Reply:
> One last request: Do you mind moving this up and pass `ctx` it into `n.instancesAPI.Resync`? It doesn't affect correctness (since the context won't be cancelled till later), but if we ever change the behavior of `Resync` to spawn any go routines, it will ensure that those are cancelled when the node is removed.

ok